### DISCUSSION TITLE
Git Branches and ConfyUI Managers Updates

### DISCUSSION THREAD

**Discussion by darkpixel** [Comfy-Org/ComfyUI-Manager#346] [Ideas]
A few days ago I wrote a new "random prompt" node with a few features I wanted that I couldn't find in any of the other custom nodes out there.  When I thought it was good enough, I tested it by installing it via Manager's "Install via GIT URL".

It installed, and then I noticed a few minor errors....so I made a few tweaks and updated my repo.

When I went back into Manager and hit "fetch all", it said there were no updates.

I dug through some code and the issues, and stumbled on to issue #202 and I think I realized why it's not updating.

My branch is named "master", not "main".  I haven't had a chance to double-check and test yet, but #202 sorta talks about wanting different branches.  I'm not sure if that's my issue, but I think it might be helpful to see a list of branches for installed modules so people can switch.  Maybe a dropdown with the output of ```git branch -r``` (with some light formatting) and a "switch branch" button?

**Comment by ltdrdata** [comment]
#202 is an issue related to the forked repository rather than the branch.

And by default, ComfyUI-Manager pull the branch specified as the `default branch` in the repository. If the default branch has changed, manager won't be handled correctly, so you'll need to clone it anew.

**Reply by darkpixel** [reply]
I guess I'll have to do some more testing on this.
I did dig through the code and found where it queries for the default branch...but it flat-out refused to update my local copy.
I do my work under ~/code/darkprompts, and when I'm ready I git push.
In the middle of my testing, you published the PR to include darkprompts in the database and then it started working correctly.  I'll just have to create a new unpublished test repo to see if I can duplicate it.
